numpy/linalg/tests/test_linalg.py

```diff
@@ -320,43 +348,87 @@ def _check_cases(func, cases):
             raise AssertionError(msg)
 
 
-class LinalgTestCase(object):
+class LinalgSquareTestCase(object):
 
     def test_sq_cases(self):
-        _check_cases(self.do, SQUARE_CASES)
+        _check_cases(self.do,
+            require=CaseFlags.square,
+            exclude=CaseFlags.generalized | CaseFlags.empty)
+
+    def test_empty_sq_cases(self):
+        _check_cases(self.do,
+            require=CaseFlags.square | CaseFlags.empty,
+            exclude=CaseFlags.generalized)
 
 
 class LinalgNonsquareTestCase(object):
 
-    def test_sq_cases(self):
-        _check_cases(self.do, NONSQUARE_CASES)
+    def test_nonsq_cases(self):
+        _check_cases(self.do,
+            require=CaseFlags.none,
+            exclude=CaseFlags.generalized | CaseFlags.square | CaseFlags.hermitian | CaseFlags.empty)
```

Comment:
My counter argument would be that it isn't important ;) If you were writing fly-by-wire software that had to be 100% correct at all times, then using an uncommon python feature would be justified. In fact, seeing such usage makes me stop and think and track down why it is important. But here I don't think it matters and I've burned up a couple of precious grey cells to no purpose. The simplest approach that works is what I would chose in this case because it makes the code that much more accessible, and likely more error free as it is easier to check.